### PR TITLE
Update the fk patch

### PR DESCRIPTION
**Pull request by seisman**
Many users have reported crashes/errors like the one shown in the screenshot:

image

I think it's because of the wrong `FFLAGS` flags.

As per the [GFortran manual](https://gcc.gnu.org/onlinedocs/gfortran/Debugging-Options.html), flags like `-ffpe-trap=overflow,invalid,denormal` enable some floating point exception traps, and is mostly used for debugging purpose. 

Removing them so that compilers can just ignore these traps.

### CODE REVIEW

**Comment by core-man** [comment]
Please update the patch file name with the latest date and the tutorial updating date.

---

**Some other minor revisions:**

- Line 58-61: Add `~/src`
```
4.  进入源码目录并编译::

        $ cd ~/src/fk/
```

Line 49-52: Add `$ cd ~/src/fk/`
```
将其放在 fk 源码目录下，然后执行如下命令即可修改源码::

            $ cd ~/src/fk/
            $ patch < fk3.3-v20190618.patch
```

**Review by core-man — APPROVED** [review]

**Comment by seisman** [comment]
> update the patch file name with the latest date

The date in the patch file name is the latest modification date of files in the fk package. So it shoudl not be changed.

**Comment by core-man** [comment]
> > update the patch file name with the latest date
> 
> The date in the patch file name is the latest modification date of files in the fk package. So it shoudl not be changed.

I see

**Comment by core-man** [comment]
I forgot one comment: we need to add a note for this revison since the original makefile contains this option.

**Comment by seisman** [comment]
> I forgot one comment: we need to add a note for this revison since the original makefile contains this option.

See 37609e6

**Comment by core-man** [comment]
Do we want to add a note for the `no-pie` issue?

**Comment by seisman** [comment]
> Do we want to add a note for the `no-pie` issue?

I don't remember what it is.

**Comment by core-man** [comment]
> > Do we want to add a note for the `no-pie` issue?
> 
> I don't remember what it is.

Okay, I will submit an issue tomorrow.

**Comment by core-man** [comment]
Some fk users also reported compilation error shown blew:

fk-make-issue

It seems that `saclib.a` is compiled without `-fpie`, so we may need to add `-no-pie` option at `SACLIB` in makefile to use it.

See https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html
> -no-pie
> Don’t produce a dynamically linked position independent executable.

Besides, I found Ubuntu 18 and CentOS 7.5 do not have this bug, but Ubuntu 20 cannot work. I guess GCC in Ubuntu 20 is configured to build with `-pie` by default (see https://stackoverflow.com/questions/34519521/why-does-gcc-create-a-shared-object-instead-of-an-executable-binary-according-to) so we need to add `-no-pie` manually, while the defaul may be `-no-pie` in Ubuntu 18 and CentOS.

File: source/fk/install.rst [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 :本节贡献者: |田冬冬|\（作者）、
              |姚家园|\（审稿）
-:最近更新日期: 2021-01-13
+:最近更新日期: 2021-03-13
```
Comment (core-man) [inline source/fk/install.rst:6]:
```suggestion
:最近更新日期: 2021-03-14
```

File: source/fk/install.rst [hunk 2]
```diff
@@ -30,6 +30,8 @@
     -   :file:`Makefile` 中需要添加 ``FC=gfortran`` 指定使用 gfortran 编译器
     -   :file:`Makefile` 中需要给 ``FFLAGS`` 加上 ``-ffixed-line-length-none``
         使得 Fortran 一行可以超过 72 字符
+    -   :file:`Makefile` 中 ``FFLAGS`` 里的 ``-ffpe-trap=overflow,invalid,denormal``
+        参数需要删除
     -   :file:`Makefile` 中第 6—7 行被注释了，所以 ``fk`` 无法使用 SAC 提供的
         滤波功能。若 SAC 已安装，则可以将第 6—7 行的行首注释符号 ``#`` 去掉
```
Comment (core-man) [inline source/fk/install.rst:36]:
```suggestion
    -   :file:`Makefile` 中第 6—7 行被注释了，所以 ``fk`` 无法使用 SAC 提供的
        滤波功能。若 SAC 已安装，则可以将第 6—7 行的行首注释符号 ``#`` 去掉。同时在
        ``SACLIB`` 中增加 ``-no-pie`` 参数。
```

File: source/fk/install.rst [hunk 3]
```diff
@@ -30,8 +30,11 @@
     -   :file:`Makefile` 中需要添加 ``FC=gfortran`` 指定使用 gfortran 编译器
     -   :file:`Makefile` 中需要给 ``FFLAGS`` 加上 ``-ffixed-line-length-none``
         使得 Fortran 一行可以超过 72 字符
+    -   :file:`Makefile` 中 ``FFLAGS`` 里的 ``-ffpe-trap=overflow,invalid,denormal``
+        参数需要删除
     -   :file:`Makefile` 中第 6—7 行被注释了，所以 ``fk`` 无法使用 SAC 提供的
-        滤波功能。若 SAC 已安装，则可以将第 6—7 行的行首注释符号 ``#`` 去掉
+        滤波功能。若 SAC 已安装，则可以将第 6—7 行的行首注释符号 ``#`` 去掉。同时给
+        ``SACLIB`` 加上 ``-no-pie`` 参数。
```
Comment (core-man) [inline source/fk/install.rst:37]: CentOS 7 cannot use `-no-pie`.
```suggestion
        滤波功能。若 SAC 已安装，则可以将第 6—7 行的行首注释符号 ``#`` 去掉。同时给
        ``SACLIB`` 加上 ``-no-pie`` 参数（若操作系统无法使用该参数，手动删除即可）。
```

**Comment by core-man** [comment]
I have no other comments.

**Pull request merged**